### ISSUE TITLE
[BUG] panic: assertion failed: leader.elected.at.same.term.0 when master.port.grpc is set explicitly

### ISSUE THREAD

**Issue by alebcay** [chrislusf/seaweedfs#2847]
**Describe the bug**
Starting in SeaweedFS 2.96, when running `weed server` with an argument `-master.port.grpc=<some open port>`, server startup crashes after cluster creation/initialization and leader change event.

This appears to be new to 2.96, I did not run into this issue with the same command on 2.95.

**System Setup**
- `weed server -dir=$(pwd) -master.port.grpc=12345`
- OS version: macOS 12.2.1 (21D62)
- output of `weed version`: `version 30GB 2.96 0490ee87efa25ecb77b71d5f7b08d539195db8c4 darwin amd64`
- if using filer, show the content of `filer.toml`: not using `filer`

**Expected behavior**
I expected the server to start up correctly, just as it would without the flag - just with a different port to be used for the master GRPC port.

**Additional context**
```
❯ weed server -dir=$(pwd) -master.port.grpc=12345
I0329 12:27:13 71664 master.go:232] current: 192.168.68.106:9333 peers:
I0329 12:27:13 71664 file_util.go:23] Folder /Users/caleb/test Permission: -rwxr-xr-x
I0329 12:27:13 71664 file_util.go:23] Folder /Users/caleb/test Permission: -rwxr-xr-x
I0329 12:27:13 71664 master.go:232] current: 192.168.68.106:9333 peers:192.168.68.106:9333.12345
I0329 12:27:13 71664 master_server.go:122] Volume Size Limit is 30000 MB
I0329 12:27:13 71664 master.go:143] Start Seaweed Master 30GB 2.96 0490ee87efa25ecb77b71d5f7b08d539195db8c4 at 192.168.68.106:9333
I0329 12:27:13 71664 volume_grpc_client_to_master.go:41] checkWithMaster 192.168.68.106:9333: get master 192.168.68.106:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 192.168.68.106:12345: connect: connection refused"
I0329 12:27:13 71664 raft_server.go:80] Starting RaftServer with 192.168.68.106:9333
I0329 12:27:13 71664 raft_server.go:129] current cluster leader:
I0329 12:27:15 71664 volume_grpc_client_to_master.go:41] checkWithMaster 192.168.68.106:9333: get master 192.168.68.106:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 192.168.68.106:12345: connect: connection refused"
I0329 12:27:17 71664 volume_grpc_client_to_master.go:41] checkWithMaster 192.168.68.106:9333: get master 192.168.68.106:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 192.168.68.106:12345: connect: connection refused"
I0329 12:27:18 71664 volume_grpc_client_to_master.go:41] checkWithMaster 192.168.68.106:9333: get master 192.168.68.106:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 192.168.68.106:12345: connect: connection refused"
I0329 12:27:20 71664 volume_grpc_client_to_master.go:41] checkWithMaster 192.168.68.106:9333: get master 192.168.68.106:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 192.168.68.106:12345: connect: connection refused"
I0329 12:27:22 71664 volume_grpc_client_to_master.go:41] checkWithMaster 192.168.68.106:9333: get master 192.168.68.106:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 192.168.68.106:12345: connect: connection refused"
I0329 12:27:24 71664 volume_grpc_client_to_master.go:41] checkWithMaster 192.168.68.106:9333: get master 192.168.68.106:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 192.168.68.106:12345: connect: connection refused"
I0329 12:27:25 71664 volume_grpc_client_to_master.go:41] checkWithMaster 192.168.68.106:9333: get master 192.168.68.106:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 192.168.68.106:12345: connect: connection refused"
I0329 12:27:27 71664 volume_grpc_client_to_master.go:41] checkWithMaster 192.168.68.106:9333: get master 192.168.68.106:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 192.168.68.106:12345: connect: connection refused"
I0329 12:27:29 71664 volume_grpc_client_to_master.go:41] checkWithMaster 192.168.68.106:9333: get master 192.168.68.106:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 192.168.68.106:12345: connect: connection refused"
I0329 12:27:31 71664 volume_grpc_client_to_master.go:41] checkWithMaster 192.168.68.106:9333: get master 192.168.68.106:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 192.168.68.106:12345: connect: connection refused"
I0329 12:27:31 71664 master.go:176] Start Seaweed Master 30GB 2.96 0490ee87efa25ecb77b71d5f7b08d539195db8c4 grpc server at 192.168.68.106:12345
I0329 12:27:32 71664 masterclient.go:80] No existing leader found!
I0329 12:27:32 71664 raft_server.go:146] Initializing new cluster
I0329 12:27:32 71664 master_server.go:165] leader change event:  => 192.168.68.106:9333.12345
I0329 12:27:32 71664 master_server.go:168] [ 192.168.68.106:9333.12345 ] 192.168.68.106:9333.12345 becomes leader.
panic: assertion failed: leader.elected.at.same.term.0


goroutine 77 [running]:
github.com/chrislusf/raft._assert(...)
	github.com/chrislusf/raft@v1.0.8/util.go:60
github.com/chrislusf/raft.(*server).processAppendEntriesRequest(0xc000241c20, 0xc000676dc0)
	github.com/chrislusf/raft@v1.0.8/server.go:983 +0x8c5
github.com/chrislusf/raft.(*server).leaderLoop(0xc000241c20)
	github.com/chrislusf/raft@v1.0.8/server.go:884 +0x3ba
github.com/chrislusf/raft.(*server).loop(0xc000241c20)
	github.com/chrislusf/raft@v1.0.8/server.go:609 +0x156
github.com/chrislusf/raft.(*server).Start.func1()
	github.com/chrislusf/raft@v1.0.8/server.go:470 +0x5d
created by github.com/chrislusf/raft.(*server).Start
	github.com/chrislusf/raft@v1.0.8/server.go:468 +0x1ae
```

**Comment by kmlebedev** [comment]
I think it's related
https://github.com/chrislusf/seaweedfs/issues/2833

**Comment by kmlebedev** [comment]
I add cmd to https://github.com/chrislusf/seaweedfs/blob/master/docker/compose/local-clusters-compose.yml
```
 command: "-v 4 server -ip=server1 -filer -volume.max=0 -master.volumeSizeLimitMB=1024 -volume.preStopSeconds=1 -master.port.grpc=12345 -master.peers=server2:12346"

```

and  run make 2clusters 
```
server1_1  | I0331 09:43:38     1 raft_server.go:146] Initializing new cluster
server1_1  | [raft]09:43:38.869776 [server1:9333.12345 Term:0] selfjoin and promote to leader
server1_1  | I0331 09:43:38     1 master_server.go:165] leader change event:  => server1:9333.12345
server1_1  | I0331 09:43:38     1 master_server.go:168] [ server1:9333.12345 ] server1:9333.12345 becomes leader.

```

**Comment by kmlebedev** [comment]
```
server1_1  | [raft]10:04:20.318302 [server1:12345 Term:0] selfjoin and promote to leader
server1_1  | [raft]10:04:20.318480 [server1:12345 Term:0] server.command.process
server1_1  | I0331 10:04:20     1 masterclient.go:80] No existing leader found!
server1_1  | I0331 10:04:20     1 raft_server.go:146] Initializing new cluster
server1_1  | I0331 10:04:20     1 master_server.go:165] leader change event:  => server1:12345
server1_1  | I0331 10:04:20     1 master_server.go:168] [ server1:12345 ] server1:12345 becomes leader.
server1_1  | [raft]10:04:20.320673 [server1:12345 Term:0] server.peer.add:  server1:12345 0
server1_1  | [raft]10:04:20.325023 setCommitIndex.set.result index: 1, entries index: 0
server1_1  | [raft]10:04:20.325142 [server1:12345 Term:0] commit index  1
server1_1  | [raft]10:04:20.325171 [server1:12345 Term:0] server.loop.run  leader
server1_1  | [raft]10:04:20.325200 [server1:12345 Term:0] leaderLoop.set.PrevIndex to  1
server1_1  | [raft]10:04:20.325299 [server1:12345 Term:0] server.command.process
server1_1  | [raft]10:04:20.325440 setCommitIndex.set.result index: 2, entries index: 1
server1_1  | [raft]10:04:20.325546 [server1:12345 Term:0] commit index  2
```

@chrislusf  Why is such an invalid network address with a dot used?

https://github.com/chrislusf/seaweedfs/blob/21e08986317995a89274b5fc7dad80a42006bc16/weed/pb/server_address.go#L19
```
	return ServerAddress(address + "." + strconv.Itoa(grpcPort))

```

**Comment by chrislusf** [comment]
Need a way to customize grpc port.

**Comment by kmlebedev** [comment]
> fix server address for grpcPort #2857

It works like this from https://github.com/chrislusf/seaweedfs/pull/2857 but it's a crutch

**Comment by chrislusf** [comment]
This commit daca2d22a56fb1ff6d659fe8dda2c5d3d0c65a15 fixed this issue.

```
p$ weed master -port 9001 -port.grpc=9002
I0401 20:13:20 28316 file_util.go:23] Folder /var/folders/jv/0rlhdck17jzgt7b3hcnq63mc0000gp/T/ Permission: -rwx------
I0401 20:13:20 28316 master.go:232] current: 192.168.2.7:9001 peers:
I0401 20:13:20 28316 master_server.go:122] Volume Size Limit is 30000 MB
I0401 20:13:20 28316 master.go:143] Start Seaweed Master 30GB 2.96  at 192.168.2.7:9001
I0401 20:13:20 28316 raft_server.go:80] Starting RaftServer with 192.168.2.7:9001
I0401 20:13:20 28316 raft_server.go:129] current cluster leader:
I0401 20:13:38 28316 master.go:176] Start Seaweed Master 30GB 2.96  grpc server at 192.168.2.7:9002
I0401 20:13:39 28316 masterclient.go:85] No existing leader found!
I0401 20:13:39 28316 raft_server.go:146] Initializing new cluster
I0401 20:13:39 28316 master_server.go:165] leader change event:  => 192.168.2.7:9001.9002
I0401 20:13:39 28316 master_server.go:168] [ 192.168.2.7:9001.9002 ] 192.168.2.7:9001.9002 becomes leader.
I0401 20:13:43 28316 master_grpc_server.go:278] + client master@192.168.2.7:9001.9002

```
